File: backend/Domain/Entities/Asset.cs
using IndustrialAssets.Domain.Enums;
namespace IndustrialAssets.Domain.Entities;
public sealed class Asset
{
    public Guid Id { get; set; }
    public required string Code { get; set; }
    public required string Name { get; set; }
    public string Description { get; set; } = string.Empty;
    public AssetType Type { get; set; }
    public required string Manufacturer { get; set; }
    public required string Model { get; set; }
    public required string SerialNumber { get; set; }
    public AssetStatus Status { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
}

File: frontend/src/types/asset.ts
export const assetTypes = ['Motor', 'Pump', 'Conveyor', 'Tank', 'Robot', 'Machine', 'Other'] as const;
export const assetStatuses = ['Operating', 'Stopped', 'Maintenance', 'Alarm', 'Offline'] as const;
export const statusLabels = { Operating: 'Operando', Stopped: 'Parado', Maintenance: 'Manutenção', Alarm: 'Alarme', Offline: 'Offline' } as const;

export type AssetType = (typeof assetTypes)[number];
export type AssetStatus = (typeof assetStatuses)[number];

export interface Asset {
  id: string;
  code: string;
  name: string;
  description: string;
  type: AssetType;
  manufacturer: string;
  model: string;
  serialNumber: string;
  status: AssetStatus;
  createdAt: string;
}

export type AssetInput = Omit<Asset, 'id' | 'createdAt'>;
